### ISSUE TITLE
Tensor Device Inconsistencies in Projected Gradient Descent Algorithm.

### ISSUE THREAD

**Issue by SignedQiu** [Trusted-AI/adversarial-robustness-toolbox#2123]
**Describe the bug**
Then tensors are not on the same device.

**To Reproduce**
1. Go to 'art/evasion/projected_gradient_descent/'
2. Click on 'projected_gradient_descent_pytorch.py'
3. Scroll down to 'line 272'

**Expected behavior**
Expected all tensors to be on the same device.

**Screenshots**
Change
![image](https://user-images.githubusercontent.com/66827919/234063217-c66a7f70-e3a6-440b-9990-c0b98439eca7.png)
To
![image](https://user-images.githubusercontent.com/66827919/234063274-25d7052c-41ad-49eb-87f3-3783b3d95d14.png)

**System information (please complete the following information):**
 - OS Linux
 - Python version 3.7.9
 - ART version or commit number 1.14.0
 - Pytorch 1.11.0+cu113

**Comment by beat-buesser** [comment]
Hi @SignedQiu Thank you very much for reporting this bug! Would you be interested to open a pull request for your solution?

**Comment by SignedQiu** [comment]
Sure! I'll do it latter.

**Comment by SignedQiu** [comment]
> Hi @SignedQiu Thank you very much for reporting this bug! Would you be interested to open a pull request for your solution?
Sure! I'll do it latter.
